### PR TITLE
[FEATURE]: added AppCenterCleaner and upgrade deps

### PR DESCRIPTION
**Pull request by Jamyth**
Feature:

- Added AppCenterCleaner and related types & APIs

Chores:

- upgrade deps

### CODE REVIEW

**Comment by dionshihk** [comment]
Thanks @Jamyth for the new feature. I have 2 suggestions for the code:

- Instead of `dryMode`, it is better to display a confirmation message of what projects will be removed, and what will be kept, then ask user to confirm (by entering `yes` / `no`).

- `include` / `exclude` mode seems no need, given the following option:

```TypeScript
type ProjectFilter = (appCenterProjects: string[]) => string[];
```

What about just let the filter return the project names to be removed.
In other words, the user implements the inclusive/exclusive algorithm inside his `ProjectFilter` function.
